core/model/modx/processors/browser/file/download.class.php

```diff
@@ -34,9 +35,22 @@ public function getObjectUrl() {
     }
 
     public function download() {
-        $fileobj = $this->source->fileHandler->make($this->source->getBasePath() . $this->getProperty('file'));
+        try {
+            /** @var League\Flysystem\Filesystem $filesystem */
+            $filesystem = $this->source->getFilesystem();
+            if ($data = $filesystem->getMetadata($this->getProperty('file'))) {
+                $name = array_pop(explode(DIRECTORY_SEPARATOR, $data['path']));
+                header('Content-type: ' . $data['mimetype']);
+                header('Content-Disposition: attachment; filename=' . $name);
+                header('Content-Length: ' . $data['size']);
 
-        $fileobj->download();
+                exit($filesystem->read($data['path']));
```

Comment:
To prevent E_WARNING notifications, before `exit()` add `@session_write_close();` to close the session.